### DISCUSSION TITLE
How to place the table of contents under a specific heading with rehype-toc?

### DISCUSSION THREAD

**Discussion by mmoser18** [remarkjs/remark#1348] [Q&A]
I am trying to use rehype-toc to generate me the table of contents of a help-page (which is converted from markdown using remark, remark-rehype etc.). It seems I need to use _rehype_-toc (instead of _remark_-toc) to generate correct and working cross-reference-links (i.e. links from the TOC to the referenced chapters that actually work).

However _rehype_-toc always places the TOC at the very beginning of the generated document. With _remark_-toc I had been able to specify a specific heading under which the TOC was to be placed. I would like to keep it that way, i.e. having my TOC only _after_ an introductory chapter, not at the very beginning. I tried to read (and understand...) the misc. readme's, examples and discussions re. rehype-toc but found no such option nor any fitting example. Can that be done or is it hard-coded in rehype-toc that the generated TOC is prepended to the document?

**Comment by ChristianMurphy** [comment]
Welcome @mmoser18! 👋 
`rehype-toc` has several versions, and all community maintained.
The author of the plugin you are using may be able to answer your question
* https://github.com/JS-DevTools/rehype-toc
* or https://github.com/Microflash/rehype-toc

It sounds a bit like you want something closer to https://github.com/stefanprobst/rehype-extract-toc
which puts the table of contents as vfile metadata and you could create your own rehype plugin (https://unifiedjs.com/learn/) to insert it to the spot you want it to go in your document(s)

**Reply by mmoser18** [reply]
@JS-DevTools/rehype-toc puts the TOC at the beginning. I don't see any option to change that. 
@Microflash/rehype-toc has a very nice feature, namely that one can put a placeholder `[[toc]]` into the markdown to mark the spot where the plugin should put the TOC. That's in principle exactly what I was seeking and the plugin also puts the TOC there, however, instead of a decent table of contents it generates is a flat, non-hierarchical list of headings. ||-(   And I see no option to change that, so: useless for me!
Sigh! I had hoped to get away without programming this myself... :-(
But thanks for pointing tho these two projects! I'll ask there whether they couldn't add such a feature.
